src/braket/circuits/parameterizable.py

```diff
@@ -0,0 +1,36 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+from __future__ import annotations
+
+from typing import List, Union
+
+from braket.circuits.free_parameter import FreeParameter
+
+
+class Parameterizable:
+    """A parameterized object is the definition of an object that can take in FreeParameters"""
+
+    def __init__(self):
+        self._parameters = list()
+
+    @property
+    def parameters(self) -> List[Union[FreeParameter, float]]:
```

Comment:
Also, not sure if it's useful to return parameters if they're already bound; after all the symbol is gone, and it's not clear what the number even means.

Reply:
The parametric circuit retains the parameterized objects. The new circuit will have a copy of objects now parameterized or a copy of the same type (ex. a copy of an Rx gate) with the parameters fixed.

Reply:
In that case, it's probably more useful to return a dict

Reply:
I don't follow why it's more useful as a dict?

Reply:
Mainly around the confusion of a heterogeneous list, where the order of the bound values isn't really meaningful. In fact, the order of this list isn't particularly meaningful either.

Reply:
I guess it depends on the key and if the "keys" (or what's used as the keys) can be duplicated or not in this list.